server.js

```diff
@@ -3179,22 +3179,53 @@ cache(function (data, match, sendBadge, request) {
     }
     try {
       var data = JSON.parse(buffer);
-        if (info === 'v') {
-          var vdata = versionColor(data);
-          badgeData.text[1] = vdata.version;
-          badgeData.colorscheme = vdata.color;
-          sendBadge(format, badgeData);
-        } else if (info == 'l') {
-          var license = data.info.license;
-          badgeData.text[0] = 'license';
-          if (license == null) {
-            badgeData.text[1] = 'Unknown';
-          } else {
-            badgeData.text[1] = license;
-            badgeData.colorscheme = 'blue';
-          }
-          sendBadge(format, badgeData);
+      if (info === 'v') {
+        var vdata = versionColor(data);
+        badgeData.text[1] = vdata.version;
+        badgeData.colorscheme = vdata.color;
+        sendBadge(format, badgeData);
+      } else if (info == 'l') {
+        var license = data.info.license;
+        badgeData.text[0] = 'license';
+        if (license == null) {
+          badgeData.text[1] = 'Unknown';
+        } else {
+          badgeData.text[1] = license;
+          badgeData.colorscheme = 'blue';
         }
+        sendBadge(format, badgeData);
+      }
+    } catch(e) {
+      badgeData.text[1] = 'invalid';
+      sendBadge(format, badgeData);
+    }
+  });
+}));
+
+// Docker Hub stars integration.
+camp.route(/^\/docker\/stars\/([^\/]+)\/([^\/]+)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var user = match[1];  // eg, _ or mashape
+  var repo = match[2];  // eg, mockbin
+  var format = match[3];
+  // Convert underscore to library for api path
+  var baseURL = 'http://docker.cloudbrain.io/';
```

Comment:
That would require more code inside shields itself, it's pretty simple for the stars but the pulls data comes from the website. Of course it could be done but I don't have the time plus I already built the API for our own internal metrics. If it would help I could put a link to the [source code](https://gist.github.com/montanaflynn/957add5908beb3f676cf) (it's Golang) powering the service as a code comment above the implementations.